dll/win32/shell32/CDefView.cpp

```diff
@@ -2215,21 +2240,25 @@ LRESULT CDefView::OnContextMenu(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &b
         m_ListView.ClientToScreen(&pt);
     }
 
+    CComPtr<ICommDlgBrowser2> pcdb2;
+    if (m_pCommDlgBrowser && SUCCEEDED(m_pCommDlgBrowser->QueryInterface(IID_PPV_ARG(ICommDlgBrowser2, &pcdb2))))
+        pcdb2->Notify(static_cast<IShellView*>(this), CDB2N_CONTEXTMENU_START);
+
     // This runs the message loop, calling back to us with f.e. WM_INITPOPUP (hence why m_hContextMenu and m_pCM exist)
     uCommand = TrackPopupMenu(m_hContextMenu,
                               TPM_LEFTALIGN | TPM_RETURNCMD | TPM_LEFTBUTTON | TPM_RIGHTBUTTON,
                               pt.x, pt.y, 0, m_hWnd, NULL);
-    if (uCommand == 0)
-        return 0;
-
     if (uCommand >= DVIDM_ARRANGESORT_FIRST && uCommand <= DVIDM_ARRANGESORT_LAST)
-        return SendMessage(WM_COMMAND, uCommand, 0);
-
-    if (uCommand == FCIDM_SHVIEW_OPEN && OnDefaultCommand() == S_OK)
-        return 0;
-
-    InvokeContextMenuCommand(m_pCM, MAKEINTRESOURCEA(uCommand - CONTEXT_MENU_BASE_ID), &pt);
+    {
+        SendMessage(WM_COMMAND, uCommand, 0);
```

Comment:
You confirm that we don't actually want to return from `CDefView::OnContextMenu` what this SendMessage would give us?

Reply:
This is for WM_CONTEXTMENU and according to MSDN the return value has no meaning.